### DISCUSSION TITLE
Adding another region OSM data into existing vahalla instance

### DISCUSSION THREAD

**Discussion by NitinMehra0019** [valhalla/valhalla#4580] [Q&A] [answered]
Hi there, I have a running Valhalla instance on my local machine with some existing OSM data now I want to inject new region OSM data into it so my question is how can I do this in running instance or do I need to remove the existing one and then re-do whole process again with new data.

Any help will be appreciated, thank you

**Comment by nilsnolde** [comment, marked as answer]
If you want to “add” data, then yes, you’ll need to do the whole process from scratch. If it’s adjacent areas I’d recommend to not provide multiple OSM files and rather download the bigger “parent” OSM file and extract from it whatever region you want with osmium or so.

**Reply by NitinMehra0019** [reply]
Area is not adjacent so need to build again, appreciate your quick response

**Comment by nilsnolde** [comment]
If it’s not adjacent, ie you wouldn’t expect to route from one area to the other, you _could_ build only the new area and merge the 2 graphs. Nothing wrong with that. It’s literally just merging the graph directories and re-building a new tarball.

**Reply by NitinMehra0019** [reply]
Alright, thanks @nilsnolde, I have one more concern, let see a scenario alright let say one valhalla instance have some routing graph of data 1 and another instance have data of data 2 assume a user request for a route which has a path from data 1 to data 2 so my question is is this possible to have 2 or more valhalla instance communication in these scenario

**Reply by nilsnolde** [reply]
nope, must be all in one
